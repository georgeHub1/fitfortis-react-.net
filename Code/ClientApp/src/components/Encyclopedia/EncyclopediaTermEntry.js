import React, { Component } from 'react';
import {
  Alert,
  Icon,
  Button,
  notification,
  Menu,
  Dropdown
} from 'antd';
import { connect } from 'react-redux';
import ReactLoading from 'react-loading';
import { injectIntl, addLocaleData, IntlProvider } from 'react-intl';
import CustomCard from '../Common/CustomCard.js';
import styles from './styles.module.less';
import { SanitizeHTML } from '../SanitizeHTML/SanitizeHTML';
import { fetchEncyclopediaDetail } from '../../redux/encyclopedia.action';
import { ReactComponent as share } from '../../img/share.svg';
import { ReactComponent as print } from '../../img/print.svg';
import { ReactComponent as checkMark } from '../../img/greenCheckMark.svg';
import { setActiveItem } from '../../redux/item.action';
import bgBG from 'react-intl/locale-data/bg';
import enUS from 'react-intl/locale-data/en';
import ukUA from 'react-intl/locale-data/uk';
import en from 'react-intl/locale-data/en';
import { flattenMessages } from '../../utils/translation';
import messages from '../../translation';
import { localeSet } from '../../redux/locale.action';
// import auth from '../../backendServices/auth';
import SignUpPrompt from '../Common/SignUpPrompt';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';
addLocaleData([...bgBG, ...enUS, ...ukUA, ...en]);

const locale = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

let intlProvider = new IntlProvider({ locale, messages: flattenMessages(messages[locale]) });


class EncyclopediaTermEntry extends Component {
  displayName = EncyclopediaTermEntry.name;
  intervalID = 0;
  constructor (props) {
    super(props);
    this.state = {
      showAlert: false,
      loading: true,
      fullInfo: {},
      id: 0,
      open: true,
      interval: 0,
      isClicked: false,
      position: 0,
      currentPath: '',
      currentId: '',
      isToggle: false
    };
    this.model3D = React.createRef();
  }
  componentDidMount () {
    const pathSnippets = window.location.pathname
          .split('/')
          .filter(i => i);

    analyticId.firebaseAnalyticsLog('Encyclopedia_Topics_Item', {encyclopediaId: pathSnippets[1]});
    this.setState({
      ...this.state,
      loading: false,
      currentId: pathSnippets[pathSnippets.length - 1]
    });
    const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

    if (window.location.search.includes('?lang=')) {
      this.props.onUpdateLanguage(lan);
      localStorage.setItem('language', lan);
    }
    intlProvider = new IntlProvider({ locale: lan, messages: flattenMessages(messages[lan]) });
    this.props.fetchEncyclopediaDetail(pathSnippets[pathSnippets.length - 1], lan);
    fetch(`/encyclopedia/${pathSnippets[1]}?lang=${lan}`);
  }
  componentWillUnmount () {
    intlProvider = new IntlProvider({ locale: (localStorage.getItem('language') || 'en'), messages: flattenMessages(messages[(localStorage.getItem('language') || 'en')]) });
    this.props.setActiveItem(null);
    clearInterval(this.intervalID);
  }
  copyToClipboard = () => {
    const cpLink = this.getURLForShare();

    const textField = document.createElement('textarea');

    textField.innerText = cpLink;
    document.body.appendChild(textField);
    textField.select();
    document.execCommand('copy');
    textField.remove();
  };
  static getDerivedStateFromProps (nextProps, prevState) {
    if (nextProps.location) {
      if (nextProps.location.pathname !== prevState.currentPath) {
        const pathSnippets = nextProps.location.pathname
          .split('/')
          .filter(i => i);

        return {
          currentPath: nextProps.location.pathname,
          currentId: pathSnippets[pathSnippets.length - 1],
          fullInfo: {
            image: 'https://doctorsurgentcare.org/wp-content/uploads/2017/10/Doctors-Urgent-Care-70458-Pneumonia.jpg'
          }
        };
      }
    }

    return null;
  }

  getURLForShare () {
    const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

    return `${window.location.origin}${window.location.pathname}?lang=${lan}`;
  }

  shareFacebook = (url = window.location.href) => {
    analyticId.firebaseAnalyticsLog('Encyclopedia_Topics_Item_ClickShare', {shareTo: 'shareToFacebook'});
    return window.open(
      `http://facebook.com/sharer.php?s=100&p[url]=${url}`,
      'fbshare',
      'height=380,width=660,resizable=0,toolbar=0,menubar=0,status=0,location=0,scrollbars=0'
    );
  };
  shareEmail = (url = window.location.href) => {
    analyticId.firebaseAnalyticsLog('Encyclopedia_Topics_Item_ClickShare', {shareTo: 'shareViaEmail'});
    return window.open(
      `${'mailto:?cc=&subject=Share a link&body='}${url}`,
      '_self'
    );
  };
  render () {
    const { showAlert, loading } = this.state;

    const { formatMessage } = (window.location.search.includes('?lang=')) ? intlProvider.getChildContext().intl : this.props.intl;

    const { synonyms = [], title = '', description = '', symptoms = [] } = this.props.encyclopedia;

    const footerClass = (!description.trim()) ? styles.footerClass : '';
    const { isDisable } = this.props;

    const openNotification = () => {
      notification.success({
        message: formatMessage({ id: 'EncyclopediaEntry.copied' })
      });
    };
    const menu = (
      <Menu className="dropDownShareMenu">
        <Menu.Item onClick={() => this.shareFacebook(this.getURLForShare())}>
          <div className={styles.menuItemBlock}>
            <Icon type="facebook" style={{ marginRight: 10 }} />
            <span>{formatMessage({ id: 'EncyclopediaEntry.shareToFacebook' })}</span>
          </div>
        </Menu.Item>
        <Menu.Item onClick={() => this.shareEmail(this.getURLForShare())}>
          <div className={styles.menuItemBlock}>
            <Icon type="mail" style={{ marginRight: 10 }} />
            <span>{formatMessage({ id: 'EncyclopediaEntry.shareToEmail' })}</span>
          </div>
        </Menu.Item>
        <Menu.Item
          onClick={() => {
            analyticId.firebaseAnalyticsLog('Encyclopedia_Topics_Item_ClickShare', {shareTo: 'copyLink'});
            openNotification();
            this.copyToClipboard();
          }}
        >
          <div className={styles.menuItemBlock}>
            <Icon type="paper-clip" style={{ marginRight: 10 }} />
            <span>{formatMessage({ id: 'EncyclopediaEntry.copyLink' })}</span>
          </div>
        </Menu.Item>
      </Menu>
    );

    if (this.props.isLoading) {
      return (
        <div >
          <ReactLoading className={styles.center} type="spin" color="#335889"  />
        </div>
      );
    }

    return (
      <div className={`${styles.containerEncyclopediaEntry} EncyclopediaEntry`}>
      <div className={styles.EncyclopediaHeader} >
        <h1 className={styles.header}>
          {title}
        </h1>
        {showAlert && (
          <Alert
            type="error"
            message="Health Alert"
            description="Appendicitis is a life-threatening condition. If you believe you may
            have apendicitis, please contact an emergency room immediately!"
            showIcon
            closable
            style={{ marginBottom: 16 }}
          />
        )}
        <div className={styles.navigation} >
          <div className={styles.sharePrintBlock}>
            <Dropdown
              overlay={menu}
              trigger={['click']}
              placement="bottomCenter"
            >
              <Button type="primary">
                <Icon component={share} />
                {formatMessage({ id: 'EncyclopediaEntry.share' })}
              </Button>
            </Dropdown>
            <Button type="primary" onClick={() => {
              window.print();
              analyticId.firebaseAnalyticsLog('Encyclopedia_Topics_Item_ClickPrint', {});
            }}
            >
              <Icon component={print} />
              {formatMessage({ id: 'EncyclopediaEntry.print' })}
            </Button>
          </div>
        </div>
        </div>
        {isDisable ? <SignUpPrompt /> : ''}
        <div className="EncyclopediaEntryCustomCard">
          <CustomCard
          >
            {
              (synonyms.length > 0) ? (
                <div>
                  <p className={styles.customText}>{formatMessage({ id: 'EncyclopediaEntry.alsoKnownAs' })}{synonyms.join(', ')}.</p>
                </div>
              ) : null
            }
            <h1 className={`${styles.description} ${footerClass}`}>{formatMessage({ id: 'EncyclopediaEntry.overview' })}</h1>
            <div className={styles.blockWithImagesOfSick}>
              <div className={styles.maxWidth}>
                {loading ? (
                  <div className={styles.spinnerContainer}>
                    <ReactLoading type="spin" color="#fff" />
                  </div>
                ) : (
                  ''
                )}
                <SanitizeHTML html={description} />
              </div>
            </div>
            {
              (symptoms.length > 0) ? (
                <div>
                  <h1 className={`${styles.header} ${styles.description} ${footerClass}`}>{formatMessage({ id: 'EncyclopediaEntry.symptoms' })}</h1>
                  {
                    symptoms.map(symptom => (
                      <div
                        key={symptom}
                        className={styles.causeItem}
                      >
                        <div className={styles.info}>
                          <Icon component={checkMark} />
                          <span>{symptom}</span>
                        </div>
                      </div>
                    ))
                  }
                </div>
              ) : null
            }
          </CustomCard>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (({ encyclopedia: { encyclopedia, isLoading } }) => ({ encyclopedia, isLoading }));

const mapDispatchToProps = dispatch => ({
  fetchEncyclopediaDetail (...args) {
    return dispatch(fetchEncyclopediaDetail(...args));
  },
  onUpdateLanguage (args) {
    dispatch(localeSet(args));
    return Promise.resolve();
  },
  setActiveItem (...args) {
    return dispatch(setActiveItem(...args));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(EncyclopediaTermEntry));
