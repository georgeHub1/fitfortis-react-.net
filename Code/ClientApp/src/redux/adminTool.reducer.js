import {
  START_UPLOADING_IMAGE,
  UPLOADING_IMAGE_SUCCESS,
  UPLOADING_IMAGE_FAILED,
  CLEAR_MSG_VALUE,
  FETCH_NEWSFEED_LIST,
  FETCH_NEWSFEED_SUCCESS,
  FETCH_NEWSFEED_FAILED,
  FETCHING_NEWSFEED_ITEM,
  FETCHING_NEWSFEED_ITEM_SUCCESS,
  FETCHING_NEWSFEED_ITEM_FAILED,
  LIKE_NEWSFEED_ITEM,
  UNLIKE_NEWSFEED_ITEM,
  REMOVE_NEWSFEED_ITEM,
  CLEAR_NEWSFEED_LISTING,
  START_UPADATING_NEWSFEED,
  UPDATE_NEWSFEED_SUCCESS,
  UPDATE_NEWSFEED_FAILED,
  FETCH_ADMINTOOL_NEWSLIST,
  FETCH_ADMINTOOL_NEWSLIST_SUCCESS,
  FETCH_ADMINTOOL_NEWSLIST_FAILED,
  FETCH_ADMIN_USER_LIST,
  FETCH_ADMIN_USER_LIST_FAILED,
  FETCH_ADMIN_USER_LIST_SUCCESS,
  FETCH_ADMINTOOLS_REMOVED_NEWSLIST_START,
  FETCH_ADMINTOOLS_REMOVED_NEWSLIST_SUCCESS,
  FETCH_ADMINTOOLS_REMOVED_NEWSLIST_FAILED,
  START_RESTORE_NEWSFEED,
  SUCCESS_RESTORE_NEWSFEED,
  FAILED_RESTORE_NEWSFEED,
  START_DELETE_NEWSFEED,
  SUCCESS_DELETE_NEWSFEED,
  FAILED_DELETE_NEWSFEED,
  START_FETCH_ANALYTIC_DATA,
  FETCH_ANALYTIC_DATA_SUCCESS,
  FETCH_ANALYTIC_DATA_FAILED,
  START_GET_FETCH_ANALYTIC_DETAIL,
  GET_FETCH_ANALYTIC_DETAIL_SUCCESS,
  GET_FETCH_ANALYTIC_DETAIL_FAILED
} from './adminTool.action';

const getInitialState = () => ({
  newsFeeds: [],
  isLoading: false,
  isUploading: false,
  isDeleting: false,
  isRestoring: false,
  msg: null,
  newsFeedMsg: null,
  isFetching: false,
  isAnalyticChartDetailsLoading: false,
  analyticChartDetails: null,
  isNewsFeedLoaded: false,
  newsFeed: null,
  analyticData: null,
  isAnalyticFetchingData: false,
  isAdminNewsFeedLoaded: false,
  adminNewsFeeds: [],
  isAdminNewsListFetching: false,
  adminRemovedNewsFeeds: [],
  isAdminRemovedNewsfeedFetching: false,
  isAdminRemovedNewsFeedLoaded: false,
  adminUserList: [],
  isAdminUserListFetching: false,
  isAdminUserListLoaded: false
});

export default {
  [CLEAR_MSG_VALUE]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null
    };
  },
  [CLEAR_NEWSFEED_LISTING]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      newsFeeds: [],
      isNewsFeedLoaded: false
    };
  },
  [START_UPLOADING_IMAGE]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isUploading: true
    };
  },
  [UPLOADING_IMAGE_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: 'NEWSFEED_CREATED_SUCCESSFULLY',
      isUploading: false,
      adminNewsFeeds: ([action.data, ...state.adminNewsFeeds]).sort((a, b) => b.date - a.date),
      newsFeeds: ([action.data, ...state.newsFeeds])
    };
  },
  [UPLOADING_IMAGE_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isUploading: false
    };
  },
  [FETCH_ADMINTOOL_NEWSLIST]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isAdminNewsListFetching: true,
      isAdminNewsFeedLoaded: false,
      adminNewsFeeds: []
    };
  },
  [FETCH_ADMINTOOL_NEWSLIST_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isAdminNewsListFetching: false,
      isAdminNewsFeedLoaded: true,
      adminNewsFeeds: action.data || []
    };
  },
  [FETCH_ADMINTOOL_NEWSLIST_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isAdminNewsListFetching: false,
      isAdminNewsFeedLoaded: false,
      adminNewsFeeds: []
    };
  },
  [FETCH_ADMIN_USER_LIST]: (state = getInitialState(), action) => {
    return {
      ...state,
      isAdminUserListLoaded: false,
      isAdminUserListFetching: true,
      adminUserList: []
    };
  },
  [FETCH_ADMIN_USER_LIST_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      isAdminUserListLoaded: true,
      isAdminUserListFetching: false,
      adminUserList: action.data || []
    };
  },
  [FETCH_ADMIN_USER_LIST_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      isAdminUserListLoaded: false,
      isAdminUserListFetching: false,
      adminUserList: []
    };
  },
  [FETCH_NEWSFEED_LIST]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isFetching: true
    };
  },
  [FETCH_NEWSFEED_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      isFetching: false,
      newsFeeds: action.data,
      isNewsFeedLoaded: true
    };
  },
  [FETCH_NEWSFEED_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeeds: [],
      isFetching: false
    };
  },

  [FETCHING_NEWSFEED_ITEM]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeed: null,
      isFetching: true
    };
  },
  [FETCHING_NEWSFEED_ITEM_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeed: action.data,
      isFetching: false
    };
  },
  [FETCHING_NEWSFEED_ITEM_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeed: null,
      isFetching: false
    };
  },

  [START_FETCH_ANALYTIC_DATA]: (state = getInitialState(), action) => {
    return {
      ...state,
      analyticData: null,
      isAnalyticFetchingData: true
    };
  },
  [FETCH_ANALYTIC_DATA_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      analyticData: action.data,
      isAnalyticFetchingData: false
    };
  },
  [FETCH_ANALYTIC_DATA_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      analyticData: null,
      isAnalyticFetchingData: false
    };
  },
  [START_GET_FETCH_ANALYTIC_DETAIL]: (state = getInitialState()) => {
    return {
      ...state,
      isAnalyticChartDetailsLoading: true
    };
  },
  [GET_FETCH_ANALYTIC_DETAIL_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      isAnalyticChartDetailsLoading: false,
      analyticChartDetails: action.data
    };
  },
  [GET_FETCH_ANALYTIC_DETAIL_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      isAnalyticChartDetailsLoading: false,
      analyticChartDetails: null
    };
  },

  [LIKE_NEWSFEED_ITEM]: (state = getInitialState(), action) => {
    const { newsFeeds } = state;
    const index = newsFeeds.findIndex(x => x.id === action.data.NewsId);

    if (index > -1) {
      newsFeeds[index].likes += 1;
      newsFeeds[index].isLiked = true;
    }
    return {
      ...state,
      newsFeeds
    };
  },
  [UNLIKE_NEWSFEED_ITEM]: (state = getInitialState(), action) => {
    const { newsFeeds } = state;
    const index = newsFeeds.findIndex(x => x.id === action.data.NewsId);

    if (index > -1) {
      newsFeeds[index].likes -= 1;
      newsFeeds[index].isLiked = false;
    }
    return {
      ...state,
      newsFeeds
    };
  },
  [REMOVE_NEWSFEED_ITEM]: (state = getInitialState(), action) => {
    const { adminNewsFeeds, newsFeeds, adminRemovedNewsFeeds } = state;
    const index = adminNewsFeeds.findIndex(x => x.id === action.id);
    const indexNewsFeed = newsFeeds.findIndex(x => x.id === action.id);

    if (index > -1) {
      adminRemovedNewsFeeds.unshift(adminNewsFeeds[index]);
      adminNewsFeeds.splice(index, 1);
    }
    if (indexNewsFeed > -1) {
      newsFeeds.splice(indexNewsFeed, 1);
    }
    return {
      ...state,
      adminNewsFeeds,
      newsFeeds,
      adminRemovedNewsFeeds
    };
  },
  [START_UPADATING_NEWSFEED]: (state = getInitialState(), action) => {
    return {
      ...state,
      isUploading: true,
      newsFeedMsg: null
    };
  },
  [UPDATE_NEWSFEED_SUCCESS]: (state = getInitialState(), action) => {
    const { adminNewsFeeds, newsFeeds } = state;
    const index = adminNewsFeeds.findIndex(x => x.id === action.data.id);

    if (index > -1) {
      adminNewsFeeds[index] = {...adminNewsFeeds[index], ...action.data};
    }

    const indexNewsFeed = newsFeeds.findIndex(x => x.id === action.data.id);

    if (indexNewsFeed > -1) {
      newsFeeds[indexNewsFeed] = {...newsFeeds[indexNewsFeed], ...action.data};
    }
    return {
      ...state,
      isUploading: false,
      newsFeedMsg: 'UPDATE_NEWS_FEED_SUCCESS',
      adminNewsFeeds,
      newsFeeds
    };
  },
  [UPDATE_NEWSFEED_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isUploading: false
    };
  },
  [FETCH_ADMINTOOLS_REMOVED_NEWSLIST_START]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isAdminRemovedNewsfeedFetching: true,
      isAdminRemovedNewsFeedLoaded: false,
      adminRemovedNewsFeeds: []
    };
  },
  [FETCH_ADMINTOOLS_REMOVED_NEWSLIST_SUCCESS]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isAdminRemovedNewsfeedFetching: false,
      isAdminRemovedNewsFeedLoaded: true,
      adminRemovedNewsFeeds: action.data || []
    };
  },
  [FETCH_ADMINTOOLS_REMOVED_NEWSLIST_FAILED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isAdminRemovedNewsfeedFetching: false,
      isAdminRemovedNewsFeedLoaded: false,
      adminRemovedNewsFeeds: []
    };
  },
  [START_RESTORE_NEWSFEED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isRestoring: true
    };
  },
  [SUCCESS_RESTORE_NEWSFEED]: (state = getInitialState(), action) => {
    const { adminNewsFeeds, adminRemovedNewsFeeds, newsFeeds } = state;
    const index = adminRemovedNewsFeeds.findIndex(x => x.id === action.data.id);

    adminNewsFeeds.push(action.data);
    newsFeeds.push(action.data);
    if (index > -1) {
      adminRemovedNewsFeeds.splice(index, 1);
    }
    return {
      ...state,
      newsFeedMsg: 'SUCCESS_IN_RESTORE_NEWSFEED',
      isRestoring: false,
      adminNewsFeeds,
      adminRemovedNewsFeeds,
      newsFeeds
    };
  },
  [FAILED_RESTORE_NEWSFEED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: 'ERROR_IN_RESTORE_NEWSFEED',
      isRestoring: false
    };
  },
  [START_DELETE_NEWSFEED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: null,
      isDeleting: true
    };
  },
  [SUCCESS_DELETE_NEWSFEED]: (state = getInitialState(), action) => {
    const { adminRemovedNewsFeeds } = state;

    const index = adminRemovedNewsFeeds.findIndex(x => x.id === action.data.id);

    if (index > -1) {
      adminRemovedNewsFeeds.splice(index, 1);
    }
    return {
      ...state,
      newsFeedMsg: 'SUCCESS_IN_DELETE_NEWSFEED',
      isDeleting: false,
      adminRemovedNewsFeeds
    };
  },
  [FAILED_DELETE_NEWSFEED]: (state = getInitialState(), action) => {
    return {
      ...state,
      newsFeedMsg: 'ERROR_IN_DELETE_NEWSFEED',
      isDeleting: false
    };
  }
};

export const defaultAdminTool = getInitialState();
