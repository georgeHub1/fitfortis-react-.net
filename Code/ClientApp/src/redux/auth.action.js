import auth from '../backendServices/auth';
import profile from '../backendServices/profile';
import {
  setAccountData,
  setProfileBasicInformation,
  setProfileLifeStyle,
  setProfileChronicConditions,
  setProfileFamilyHistory,
  setProfileTherapyVaccine,
  setProfileAvatar,
  setUserRole
} from './profile.action.js';
import { localeSet } from './locale.action';
import { getAgeRange } from '../constants/symptomChecker';
import {
  setUserInfo
} from './symptomChecker.action';
import analyticId from '../utils/AnalyticsAndAbTests.tsx';

export const LOGIN_USER_REQUEST = 'LOGIN_USER_REQUEST';
export const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
export const LOGIN_USER_FAILURE = 'LOGIN_USER_FAILURE';

export const loginUserRequest = () => ({
  type: LOGIN_USER_REQUEST
});

export const loginUserSuccess = () => ({
  type: LOGIN_USER_SUCCESS
});

export const loginUserFailure = error => ({
  type: LOGIN_USER_FAILURE,
  error
});

export const fetchAccountMe = (token, push = () => true, lang) => (dispatch, getState) => {
  dispatch(loginUserRequest());

  return auth
    .accountMe(token, push)
    .then(({ item }) => {
      let language = item.hasOwnProperty('languageCode') ? item.languageCode : localStorage.getItem('language') || 'en';

      analyticId.firebaseAnalyticUserId(item.userId);
      if (lang) {
        language = lang;
      }
      if (window.location.search.includes('?lang=')) {
        language = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');
      }
      localStorage.setItem('language', language);
      dispatch(localeSet(language));
      dispatch(setUserRole(item.role.toLowerCase() || 'user'));
      dispatch(loginUserSuccess());
      dispatch(setAccountData(item));
      const fetchBasicInformation =  profile.profileFetchBasicInformation(item.userId, language);
      const fetchChronicConditions =  profile.profileFetchChronicConditions(item.userId, language);
      const fetchFamilyHistory =  profile.profileFetchFamilyHistory(item.userId, language);
      const fetchTherapyVaccine =  profile.profileFetchTherapyVaccine(item.userId, language);
      const fetchLifeStyle =  profile.profileFetchLifeStyle(item.userId, language);
      const fetchAvatar = profile.profileFetchAvatar(item.userId, token);

      Promise.all([fetchBasicInformation, fetchChronicConditions, fetchFamilyHistory, fetchTherapyVaccine, fetchLifeStyle, fetchAvatar])
      .then(([basicInformation = [], chronicConditions = [], familyHistory = [], therapyVaccine = [], lifestyle = [], avatar = undefined]) => {
        dispatch(setProfileBasicInformation(basicInformation));
        dispatch(setProfileChronicConditions(chronicConditions));
        dispatch(setProfileFamilyHistory(familyHistory));
        dispatch(setProfileTherapyVaccine(therapyVaccine));
        dispatch(setProfileLifeStyle(lifestyle));
        dispatch(setProfileAvatar(avatar));

        const userInfo = {
          isFemale: (item.genderIdentity === 'Female'),
          isPregnant: false,
          keyYear: item.dateOfBirth ? getAgeRange(item.dateOfBirth) : {},
          gender: item.sexAtBirth ? item.sexAtBirth.toLowerCase() : item.sexAtBirth
        };

        dispatch(setUserInfo(userInfo));
      }).catch(err => {
        throw err;
      });
    }, test => {
      dispatch(loginUserFailure());
    });
};

export const loginUser = (username, password, push, lang) => dispatch => {
  dispatch(loginUserRequest());

  return auth
    .authenticate(username, password)
    .then(token => {
      dispatch(fetchAccountMe(token, push, lang));
    }, errors => {
      dispatch(loginUserFailure(errors));
    });
};

export const confirmAccount = (token, userId, history) => dispatch => {
  dispatch(loginUserRequest());

  return auth
    .confirmation(token, userId)
    .then(({ item }) => {
      dispatch(fetchAccountMe(item.token, history.push));
    }, ({ errors }) => {
      history.push('/');
      dispatch(loginUserFailure(errors));
    });
};

