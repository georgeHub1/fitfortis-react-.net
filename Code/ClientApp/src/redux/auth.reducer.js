import {LOGIN_USER_REQUEST, LOGIN_USER_SUCCESS, LOGIN_USER_FAILURE} from './auth.action';

const getInitialState = () => ({
  loading: false,
  loggenIn: false,
  error: null
});

export default {
  [LOGIN_USER_REQUEST]: (state = getInitialState()) => ({
    ...state,
    error: null,
    loading: true
  }),

  [LOGIN_USER_SUCCESS]: (state = getInitialState()) => ({
    loading: false,
    loggenIn: true,
    error: null
  }),

  [LOGIN_USER_FAILURE]: (state = getInitialState(), action) => ({
    loading: false,
    loggenIn: false,
    error: action.error
  })

};

export const defaultAuth = getInitialState();
