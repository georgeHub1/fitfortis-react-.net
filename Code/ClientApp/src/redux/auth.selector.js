export const getAccountInfo = state => state.profile.account;
export const getUserId = state => getAccountInfo(state).userId;
export const getUserLanguage = state => state.locale.language;
