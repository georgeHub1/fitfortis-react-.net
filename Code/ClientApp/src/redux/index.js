import { createStore, applyMiddleware } from 'redux';

import reducers from './index.reducer';
import middleware from './middleware.store';

import { defaultAuth } from './auth.reducer.js';
import { defaultCharts } from './metricsChart.reducer';
import { defaultMetricsDateRange } from './metricsDateRange.reducer.js';
import { defaultMetricsMetric } from './metricsMetric.reducer.js';
import { defaultMetricsMetricData } from './metricsMetricData.reducer.js';
import { defaultEncyclopedia } from './encyclopedia.reducer.js';
import { defaultProfile } from './profile.reducer.js';
import { defaultSymptomChecker } from './symptomChecker.reducer.js';
import { defaultDocument } from './document.reducer';
import { defaultItem } from './item.reducer.js';
import { defaultAdminTool } from './adminTool.reducer';
const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);

export const initialState = {
  auth: defaultAuth,
  metricsChart: defaultCharts,
  metricsDateRange: defaultMetricsDateRange,
  metricsMetric: defaultMetricsMetric,
  metricsMetricData: defaultMetricsMetricData,
  encyclopedia: defaultEncyclopedia,
  profile: defaultProfile,
  symptomChecker: defaultSymptomChecker,
  document: defaultDocument,
  item: defaultItem,
  adminTool: defaultAdminTool
};

const createAppStore = createStoreWithMiddleware(
  reducers,
  initialState,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default createAppStore;
