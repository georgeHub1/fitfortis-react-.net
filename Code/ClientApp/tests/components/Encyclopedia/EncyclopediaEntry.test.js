import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import EncyclopediaTermEntry from '../../../src/components/Encyclopedia/EncyclopediaTermEntry.js';
import { createStore, applyMiddleware } from 'redux';

const Container = EncyclopediaTermEntry;
const mockStore = configureStore([]);

describe('EncyclopediaTermEntry', () => {
  it('renders correctly', () => {
  	let store = mockStore({
      encyclopedia: {
      	prevSearches: [],
        encyclopedia: {
          title: '',
          synonyms: '',
          conciseDescription: ''
        }
      }
    });
    store.dispatch = jest.fn();

    const wrapper = renderer
      .create(<Provider store={store}>
	        <Container />
	      </Provider>
	    )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
