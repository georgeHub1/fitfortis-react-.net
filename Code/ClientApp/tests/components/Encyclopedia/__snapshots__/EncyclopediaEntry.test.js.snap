// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncyclopediaTermEntry renders correctly 1`] = `
<div
  className="undefined EncyclopediaEntry"
>
  <div>
    <h1>
      
    </h1>
    <div>
      <div>
        <button
          className="ant-btn ant-dropdown-trigger ant-btn-primary"
          onClick={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <i
            className="anticon"
          >
            <svg
              aria-hidden={true}
              className=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              share.svg
            </svg>
          </i>
        </button>
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <i
            className="anticon"
          >
            <svg
              aria-hidden={true}
              className=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              print.svg
            </svg>
          </i>
        </button>
      </div>
    </div>
  </div>
  
  <div
    className="EncyclopediaEntryCustomCard"
  >
    <div
      className="ant-card undefined customCardContainer ant-card-bordered"
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.8)",
        }
      }
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <h1
          className="undefined undefined"
        />
        <div>
          <div>
            
            <div
              tabIndex="0"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
